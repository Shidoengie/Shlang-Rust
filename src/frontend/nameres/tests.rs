use crate::{
    frontend::{
        Parser,
        nameres::{error::NameErr, resolver::NameRes},
        nodes::{Node, NodeStream},
    },
    spans::{IntoSpanned, Span, Spanned},
    test_func,
};

fn test_nameres(body: &str) -> Result<NodeStream, Spanned<NameErr>> {
    let mut parser = Parser::new(body);
    let ast = parser.parse().unwrap();
    NameRes::resolve(ast)
}

test_func!(test_nameres,(
    simple_vars: "var a = 1; var a = 1; var a = 1;"
    var_usage:"var a = 1; var b = a;"
    undefined_var:"var a = c;"

    scoping:"var a = 10; do {var a = a;}"
    var_used_outside_scope:"do {var a = 10;}; var b = a;"
));
