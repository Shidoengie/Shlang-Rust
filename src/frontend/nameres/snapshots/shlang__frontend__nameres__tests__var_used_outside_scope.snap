---
source: src/frontend/nameres/tests.rs
expression: "test_nameres (\"do {var a = 10;}; var b = a;\")"
---
Err(
    UndefinedVar(
        "a",
    )[26,27],
)
